docs/en-us/web/custom-component.md

```diff
@@ -1,58 +1,61 @@
-# 自定义界面组件
+# Custom UI Components
 
-使用 Hippy 开发过程中，当前的能力可能无法满足一些特定场景，这时候就需要对 UI 组件进行封装或者引入一些第三方的功能来完成需求。
+In the development of using Hippy, it is possible that the builtin components won't meet the need of certain circumstances. In this case, you can augment the UI components by encapsulation or including some third-party functionalities inside the components.
 
-# 组件扩展
+# Components' Extensions
 
-扩展组件主要包括：
+Components' extensions can be achived by following:
 
-1. 扩展 `HippyView`
-2. 实现构造方法
-3. 实现设置自定义组件的 `tagName`
-4. 实现构造自定义组件的 `dom`
-5. 实现自定义组件的 `API 能力`
-6. 实现自定义组件的属性
+1. Extend the custom component from `HippyView` 
+2. Implement the class constuctor method
+3. Set `tagName` for the custom component
+4. Construct `dom` for the custom component
+5. Implement `API` for the custom component
+6. Implement or define properties for the custom component
 
-其中 `HippyView` 类，实现了一些 HippyBaseView 的接口和属性定义，在一个自定义组件中有几个比较重要的属性：
 
-* id: 每一个实例化 component 的唯一标识，默认会赋值给组件 dom 的 id 属性
-* pId: 每一个实例化 component 的父组件标识
-* tagName: 是用来区分组件的类型，也是业务侧代码使用该组件时在 `nativeName` 属性上填写的值，用来关联自定义组件的 key
-* dom: 真正挂载到document上的节点
-* props: 承载了从业务侧传递过来的属性和style的记录
+The `HippyView` class implements HippyBaseView's interface. There are several essential properties in a custom component:
```

Comment:
The `HippyView` class implements  some interfaces and properties of HippyBaseView.